### PR TITLE
Adds interface guidelines for TextInputWithTokens

### PR DESCRIPTION
**Pull request by mperrotti**

### CODE REVIEW

File: content/components/text-input-with-tokens.mdx [hunk 1]
```diff
@@ -0,0 +1,114 @@
+---
+title: Text input with tokens
+description: An input for a value that is a list
```
Comment (emilybrick) [inline content/components/text-input-with-tokens.mdx:3]:
```suggestion
description: Text input is for a value that is a list.
```

or Text input is an input for a value that is a list.

Comment (mperrotti) [reply content/components/text-input-with-tokens.mdx]: How about "A text input with tokens is an input for a value that is a list."?

File: content/components/text-input.mdx [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+<img
+  width="960"
+  role="presentation"
+  src="https://user-images.githubusercontent.com/2313998/224445197-4b71262c-e1cc-4495-88d5-e46c6474d3cc.png"
+/>
+
+A text input can be made smaller or larger depending on the context it's being used in.
+
+You could make an input smaller if it is being used in a tight space with other small elements. You could make an input larger if it's in an open space or it feels out of proportion with adjacent large elements.
```
Comment (emilybrick) [inline content/components/text-input.mdx:83]:
```suggestion
You can make an input smaller if it is being used in a tight space with other small elements. You can make an input larger if it's in an open space or feels out of proportion with adjacent large elements.
```

File: content/components/text-input.mdx [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+
+<img
+  width="960"
+  role="presentation"
+  src="https://user-images.githubusercontent.com/2313998/224445198-7217e528-ed15-4fe9-aba9-264c7bc62e1b.png"
+/>
+
+By default, a text input will be given a default width by the browser.
+
+However, you can render a text input to fill the width of it's parent element, or set the width to whatever makes sense for your use-case. When setting a custom width, consider the length of the expected value.
```
Comment (emilybrick) [inline content/components/text-input.mdx:95]:
```suggestion
However, you can render a text input to fill the width of its parent element, or set the width to whatever makes sense for your use-case. When setting a custom width, consider the length of the expected value.
```

File: content/components/text-input.mdx [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+
+<img
+  width="960"
+  role="presentation"
+  src="https://user-images.githubusercontent.com/2313998/224445201-7f81cf2c-992c-470b-b702-4b45d57c3234.png"
+/>
+
+A trailing action may be used to render a single action that affects the input. A trailing action is most commonly used for a button that clears the input.
+
+Ideally the trailing action button is an icon without a text label. It is also possible to render the button with a text label, but it should be avoided whenever possible.
```
Comment (emilybrick) [inline content/components/text-input.mdx:117]: Does it make sense to duplicate a lot of this documentation from https://github.com/primer/design/pull/428/files#diff-d064802a2190980eba4e2bc71b3155f9e213d88e9c2ca7f9ae295a26f6a0a115R117 or should we link?

File: content/components/text-input.mdx (hunk 3 above)
Comment (mperrotti) [reply content/components/text-input.mdx:95]: Oops, this PR was supposed to _only_ contain changes relevant to TextInputWithTokens. Once we merge https://github.com/primer/design/pull/428, there should be way less changes in `text-input.md`.

File: content/components/text-input-with-tokens.mdx (hunk 1 above)
Comment (mperrotti) [reply content/components/text-input-with-tokens.mdx]: Updated and dropped the "A"

File: content/components/text-input-with-tokens.mdx [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+
+You may choose any of Primer's token components, but you cannot mix different token components in a single input. You may also choose the size of the tokens and whether or not the token's remove button is shown.
+
+## Interactions
+
+### Move focus between tokens
+
+<Box display="flex" flexDirection={['column', 'column', 'column', 'column', 'row']} sx={{gap: 3}}>
+  <Box flex={1}>
+    When tabbing to the input, the `<input />` element receives focus. Then, when the text cursor is at the beginning of
```
Comment (langermank) [inline content/components/text-input-with-tokens.mdx:79]: Somehow the code comment renders an input 🤔 was this intentional?
image

Comment (langermank) [reply content/components/text-input-with-tokens.mdx:79]: Also, these videos are really hard to see. I think maybe recording just part of the story might make it more visible

**Review by langermank — COMMENTED** [review]
Few comments about the videos, otherwise looks great!

**Pull request merged**